lib/src/base_client.dart

```diff
@@ -23,15 +21,15 @@ abstract class BaseClient implements Client {
   /// can be a [Uri] or a [String].
   ///
   /// For more fine-grained control over the request, use [send] instead.
-  Future<Response> head(url, {Map<String, String> headers}) =>
-    _sendUnstreamed("HEAD", url, headers);
+  FutureOr<Response> head(url, {Map<String, String> headers}) =>
+    send(new Request.head(_uri(url), headers: headers));
```

Comment:
How about the named constructors take dynamic but the default takes a Uri? If all of them take dynamic then there's a check on each call to `change`.